LanguageFeatures/Patterns/variable_A02_t01.dart

```diff
@@ -60,7 +79,26 @@ String testList(List l, [bool doTypeTest = false]) {
   }
 }
 
-String testMap(Map m) {
+String testList2(List l) {
+  return switch (l) {
+    case [var a, final b] => "[$a, $b]";
+    default => "default";
+  };
+}
+
+String testList3(List l, [bool doTypeTest = false]) {
+  if (l case [var a, final b]) {
+    if (doTypeTest) {
+      a.expectStaticType<Exactly<int>>();
```

Comment:
Same issue as in LanguageFeatures/Patterns/variable_A02_t01.dart.

Reply:
Here I'd expect the elements to have type `dynamic`, based on `l` having the type `List<dynamic>`, plus [this sentence](https://github.com/dart-lang/language/blob/master/accepted/future-releases/0546-patterns/feature-specification.md#:~:text=else%20if%20m%20implements%20list%3Ct%3E%20for%20some%20t%20then%20e%20is%20t.).